packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -601,10 +607,25 @@ export class DataGrid extends FoundationElement {
     };
 
     private updateRowIndexes = (): void => {
-        const newGridTemplateColumns =
-            this.gridTemplateColumns === undefined
-                ? this.generatedGridTemplateColumns
-                : this.gridTemplateColumns;
+        let newGridTemplateColumns = this.gridTemplateColumns;
+
+        if (newGridTemplateColumns === undefined) {
+            // try to generate columns based on manual rows
+            if (this.generatedGridTemplateColumns === "" && this.rowElements.length > 0) {
+                const firstRow = this.rowElements[0] as DataGridRow;
+
+                let templateColumns: string = "";
+                firstRow.cellElements.forEach(() => {
+                    templateColumns = `${templateColumns}${
+                        templateColumns === "" ? "" : " "
+                    }${"1fr"}`;
+                });
```

Comment:
This logic could be simplified by using the [CSS `repeat` function](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat()). 

You would just need a total count of how many cells are in `firstRow` and then could set the `templateColumns` variable to `repeat(${numCells}, 1fr)`––no need for whitespace logic, etc.

Reply:
There are other variables at play when generating template columns so doesn't fit well in the template.  This is a fallback if there are no column-definitions, no row data, etc..

Reply:
Ahh, I see! Thanks for clarifying!